pkg/apis/v1beta1/ec2nodeclass_status.go

```diff
@@ -68,4 +70,23 @@ type EC2NodeClassStatus struct {
 	// InstanceProfile contains the resolved instance profile for the role
 	// +optional
 	InstanceProfile string `json:"instanceProfile,omitempty"`
+	// Conditions contains signals for health and readiness
```

Comment:
We should also add a section for this in our docs https://karpenter.sh/docs/concepts/nodeclasses/#statusinstanceprofile